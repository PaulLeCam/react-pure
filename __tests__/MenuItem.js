import React from 'react';
import TestRenderer from 'react-test-renderer';

import MenuItem from '../src/MenuItem';

describe('MenuItem', () => {
  it('renders a li with class `pure-menu-item` by default', () => {
    const renderer = TestRenderer.create(<MenuItem className="my-item" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-menu-selected` class if the `selected` property is set', () => {
    const renderer = TestRenderer.create(<MenuItem selected />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-menu-disabled` class if the `disabled` property is set', () => {
    const renderer = TestRenderer.create(<MenuItem disabled />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-menu-has-children` class if the `hasChildren` property is set', () => {
    const renderer = TestRenderer.create(<MenuItem hasChildren />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-menu-allow-hover` class if the `allowHover` property is set', () => {
    const renderer = TestRenderer.create(<MenuItem allowHover />);
    expect(renderer.toJSON).toMatchSnapshot();
  });
});
