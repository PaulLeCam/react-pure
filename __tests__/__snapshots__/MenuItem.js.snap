// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem adds the \`pure-menu-allow-hover\` class if the \`allowHover\` property is set 1`] = `[Function]`;

exports[`MenuItem adds the \`pure-menu-disabled\` class if the \`disabled\` property is set 1`] = `[Function]`;

exports[`MenuItem adds the \`pure-menu-has-children\` class if the \`hasChildren\` property is set 1`] = `[Function]`;

exports[`MenuItem adds the \`pure-menu-selected\` class if the \`selected\` property is set 1`] = `[Function]`;

exports[`MenuItem renders a li with class \`pure-menu-item\` by default 1`] = `[Function]`;
